code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -0,0 +1,372 @@
+
+	if(IC.w_class > src.w_class)
+		to_chat(user, "<span class='warning'>\The [IC] is way too big to fit into \the [src].</span>")
+		return FALSE
+
+	var/total_part_size = get_part_size()
+	var/total_complexity = get_part_complexity()
+
+	if((total_part_size + IC.size) > max_components)
+		to_chat(user, "<span class='warning'>You can't seem to add the '[IC]', as there's insufficient space.</span>")
+		return FALSE
+	if((total_complexity + IC.complexity) > max_complexity)
+		to_chat(user, "<span class='warning'>You can't seem to add the '[IC]', since this setup's too complicated for the case.</span>")
+		return FALSE
+
+	if(!user.transferItemToLoc(IC, src))
+		return FALSE
+
+	IC.assembly = src
+
+	return TRUE
+
+/obj/item/device/electronic_assembly/afterattack(atom/target, mob/user, proximity)
+	if(proximity)
+		var/scanned = FALSE
+		for(var/obj/item/integrated_circuit/input/sensor/S in contents)
+			S.set_pin_data(IC_OUTPUT, 1, weakref(target))
+			S.check_then_do_work()
+			if(S.scan(target))
+				scanned = TRUE
+		if(scanned)
+			visible_message("<span class='notice'>\The [user] waves \the [src] around [target].</span>")
+
+/obj/item/device/electronic_assembly/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/integrated_circuit))
+		if(!user.canUnEquip(I))
+			return FALSE
+		if(add_circuit(I, user))
+			to_chat(user, "<span class='notice'>You slide \the [I] inside \the [src].</span>")
+			playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+			interact(user)
+			return TRUE
+	else if(istype(I, /obj/item/crowbar))
+		playsound(get_turf(src), 'sound/items/Crowbar.ogg', 50, 1)
+		opened = !opened
+		to_chat(user, "<span class='notice'>You [opened ? "opened" : "closed"] \the [src].</span>")
+		update_icon()
+		return TRUE
+	else if(istype(I, /obj/item/device/integrated_electronics/wirer) || istype(I, /obj/item/device/integrated_electronics/debugger) || istype(I, /obj/item/screwdriver))
+		if(opened)
+			interact(user)
+		else
+			to_chat(user, "<span class='warning'>\The [src] isn't opened, so you can't fiddle with the internal components.  \
+			Try using a crowbar.</span>")
+	else if(istype(I, /obj/item/stock_parts/cell))
+		if(!opened)
+			to_chat(user, "<span class='warning'>\The [src] isn't opened, so you can't put anything inside.  Try using a crowbar.</span>")
+			return FALSE
+		if(battery)
+			to_chat(user, "<span class='warning'>\The [src] already has \a [battery] inside.  Remove it first if you want to replace it.</span>")
+			return FALSE
+		var/obj/item/stock_parts/cell = I
+		user.transferItemToLoc(I, loc)
+		cell.forceMove(src)
+		battery = cell
+		playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You slot \the [cell] inside \the [src]'s power supplier.</span>")
+		interact(user)
+		return TRUE
+	else
+		return ..()
+
+/obj/item/device/electronic_assembly/attack_self(mob/user)
+	if(!check_interactivity(user))
+		return
+	if(opened)
+		interact(user)
```

Comment:
Shouldn't there be a return here ?

Reply:
no need to return. If assembly is opened, it opens both construction and interaction windows.